src/auth/cephx/CephxServiceHandler.cc

```diff
@@ -139,6 +139,12 @@ int CephxServiceHandler::handle_request(bufferlist::iterator& indata, bufferlist
 
       if (!key_server->get_service_caps(entity_name, CEPH_ENTITY_TYPE_MON, caps)) {
         ldout(cct, 0) << " could not get mon caps for " << entity_name << dendl;
+        ret = -EACCES;
+      }else{
+        if (!caps.caps.length()){
+          ldout(cct,0) << "mon caps null for " << entity_name << dendl;
```

Comment:
test this error message?  use "client.bug" in test.sh, this will return -EACCES, client will out "Permission denied", actually we have tested this case.